src/Core/Services/VaultTimeoutService.cs

```diff
@@ -34,7 +34,7 @@ public class VaultTimeoutService : IVaultTimeoutService
             ITokenService tokenService,
             IPolicyService policyService,
             IKeyConnectorService keyConnectorService,
-            Action<bool> lockedCallback,
+            Func<Tuple<string, bool>, Task> lockedCallback,
```

Comment:
I think that named tuples are better because one can know what the parameters are inline:
`Func<(string userId, bool userInitiated), Task> lockedCallback`

Reply:
Fixed in #1830